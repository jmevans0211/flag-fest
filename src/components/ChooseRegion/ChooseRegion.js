import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './ChooseRegion.scss'

class ChooseRegion extends Component {
  constructor() {
    super();
    this.state = {
      region: '',
      tenLimit: false,
    }
  }

  handleClick = (e, region) => {
    if (region === 'africa') {
      this.setState({ region: 'Africa'})
    }
  }

  render() {
    console.log(this.state)
    return (
      <main>
        <h1>Choose a Region</h1>
        <img id="africa" className="region" onClick={(e) => this.handleClick(e, 'africa')} src="https://svgsilh.com/svg/151640.svg" />
        <img id="europe" className="region" src="https://svgsilh.com/svg/151641.svg" />
        <img id="asia-aus" className="region" src="https://svgsilh.com/svg/307197.svg" />
        <img id="americas" className="region" src="https://svgsilh.com/svg_v2/714733.svg" />
        <div className="flag-amount-container">
          <h4>Choose How Many Flags You'd Like To Be Tested On</h4>
          <button type="button">All Flags for this Region</button>
          <button type="button">Test me on 10 flags for this region</button>
        </div>
        <button type="button" className="play-button">Play!</button>
      </main>
    )
  }

}

export default ChooseRegion;