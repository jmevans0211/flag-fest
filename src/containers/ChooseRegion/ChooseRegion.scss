.choose-region-main {
  text-align: center;
}

.choose-region-heading {
  font-size: 100px;
}

.region {
  cursor: pointer;
  margin: 0 30px;
  width: 150px;
}

.region-button-container {
  display: flex;
  justify-content: space-around;
  padding: 0 60px;
}

.region-img-p {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}