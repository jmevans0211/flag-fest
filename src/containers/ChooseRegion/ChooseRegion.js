import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './ChooseRegion.scss'

class ChooseRegion extends Component {
  constructor() {
    super();
    this.state = {
      region: [],
      tenLimit: false,
    }
  }

  handleRegion = (region) => {
    if (region === 'africa') {
      this.setState({ region: ['Africa']})
    } else if (region === 'europe') {
      this.setState({ region: ['Europe']})
    } else if (region === 'asia-oceania') {
      this.setState({ region: ['Asia', 'Oceania'] })
    } else if (region === 'americas') {
      this.setState({ region: ['America']})
    }
  }

  handleFlagAmount = (amount) => {
    if (amount === 'ten') {
      this.setState({ tenLimit: true})
    } else if (amount === 'all') {
      this.setState({ tenLimit: false })
    }
  }

  render() {
    return (
      <main>
        <h1>Choose a Region</h1>
        <img className="region" onClick={() => this.handleRegion('africa')} src="https://svgsilh.com/svg/151640.svg" />
        <img className="region" onClick={() => this.handleRegion('europe')} src="https://svgsilh.com/svg/151641.svg" />
        <img className="region" onClick={() => this.handleRegion('asia-oceania')} src="https://svgsilh.com/svg/307197.svg" />
        <img className="region" onClick={() => this.handleRegion('americas')} src="https://svgsilh.com/svg_v2/714733.svg" />
        <div className="flag-amount-container">
          <h4>Choose How Many Flags You'd Like To Be Tested On</h4>
          <button type="button" onClick={() => this.handleFlagAmount('all')}>All Flags for this Region</button>
          <button type="button" onClick={() => this.handleFlagAmount('ten')}>Test me on 10 flags for this region</button>
        </div>
        <Link to='/flag-fest/play'>
          <h4 className="play-button">Play!</h4>
        </Link>
      </main>
    )
  }

}

export default ChooseRegion;