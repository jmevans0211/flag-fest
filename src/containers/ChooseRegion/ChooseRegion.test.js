import React from 'react';
import { shallow } from 'enzyme';
import { ChooseRegion } from './ChooseRegion';

describe ('ChooseRegion', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ChooseRegion countries={
      [
        {
          flag: "https://restcountries.eu/data/srb.svg",
          name: "Serbia",
          numericCode: 688,
          region: "Europe"
        },
        {
          flag: "https://restcountries.eu/data/lux.svg",
          name: "Luxembourg",
          numericCode: 442,
          region: "Europe"
        }
      ]
    }/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleRegion when region is clicked', () => {
    wrapper.instance().handleRegion = jest.fn()
    const mockEvent = { stopPropagation: jest.fn() }

    wrapper.find('img').at(0).simulate('click', mockEvent);

    expect(wrapper.instance().handleRegion).toHaveBeenCalledWith(mockEvent, 'africa')
  });

  it('should call handleRegion when region is clicked', () => {
    wrapper.instance().handleRegion = jest.fn()
    const mockEvent = { stopPropagation: jest.fn() }

    wrapper.find('img').at(1).simulate('click', mockEvent);

    expect(wrapper.instance().handleRegion).toHaveBeenCalledWith(mockEvent, 'europe')
  });
});