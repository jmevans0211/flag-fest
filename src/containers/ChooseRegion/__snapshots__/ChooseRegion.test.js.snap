// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseRegion should match snapshot 1`] = `
<main>
  <h1>
    Choose a Region
  </h1>
  <img
    className="region"
    onClick={[Function]}
    src="https://svgsilh.com/svg/151640.svg"
  />
  <img
    className="region"
    onClick={[Function]}
    src="https://svgsilh.com/svg/151641.svg"
  />
  <img
    className="region"
    onClick={[Function]}
    src="https://svgsilh.com/svg/307197.svg"
  />
  <img
    className="region"
    onClick={[Function]}
    src="https://svgsilh.com/svg_v2/714733.svg"
  />
  <img
    className="region"
    onClick={[Function]}
    src="https://svgsilh.com/svg/23512.svg"
  />
  <div
    className="flag-amount-container"
  >
    <h4>
      Choose How Many Flags You'd Like To Be Tested On
    </h4>
    <button
      onClick={[Function]}
      type="button"
    >
      All Flags for this Region
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Test me on 10 flags for this region
    </button>
  </div>
  <Link
    to="/flag-fest/play"
  >
    <h4
      className="play-button"
      onClick={[Function]}
    >
      Play!
    </h4>
  </Link>
  <Link
    to=""
  >
    <h4
      className="back-button"
    >
      Back
    </h4>
  </Link>
</main>
`;
